hack/jenkins/e2e-runner.sh

```diff
@@ -188,14 +188,18 @@ function get_gci_k8s_version() {
 }
 
 # GCI specific settings.
-# Assumes: JENKINS_GCI_IMAGE_FAMILY
+# Assumes: JENKINS_GCI_IMAGE_FAMILY & JENKINS_GCI_PROJECT
 function setup_gci_vars() {
```

Comment:
This logic is triggered whenever `JENKINS_GCI_IMAGE_FAMILY` as per [this line](https://github.com/vishh/kubernetes/blob/ad38bc4d114afbe27571109687f38b53b7432bfc/hack/jenkins/e2e-runner.sh#L241)

`JENKINS_GCI_IMAGE_FAMILY` is being set for all [GCI tests here](https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce-gci.yaml)

Am I misreading this script? To me it looks like this PR is necessary to control the image project for both master and nodes. 
If the intent is to only use this for the GCI QA jobs, then all the non qa tests must simply set `KUBE_OS_DISTRIBTION=gci` and not set the `JENKINS_GCI_IMAGE_FAMILY` env var. 
WDYT?